File: solutions/python/004_armstrong.py
def COUNT_DIGITS(NUMBER):
    digitsCount = 0
    while (NUMBER > 0):
        digitsCount += 1
        NUMBER = NUMBER // 10
    
    return digitsCount


def main(NUMBER):
    temp = NUMBER
    numberOfDigits = COUNT_DIGITS(NUMBER);
    total = 0
    while (temp > 0):
        digit = temp % 10
        total += (digit ** numberOfDigits)
        temp //= 10

    if (total == NUMBER):
        return True
    else:
        return False


number = int(input("Enter the Number: "))

if (main(number)):
    print(f"{number} IS AN ARMSTRONG")
else:
    print(f"{number} IS NOT AN ARMSTRONG")

File: solutions/cpp/004_armstrong.cpp
#include <iostream>
using namespace std;

int main() {

    long long number;
    cout << "Enter the Number: ";
    cin >> number;

    long long temp = number; // copy!!

    // calculating the number of digits first!
    int numberOfDigits = 0;
    while (temp > 0) {
        numberOfDigits++;
        temp /= 10;
    }

    temp = number;
    long long sum = 0;
    while (temp > 0) {
        int digit = temp % 10;
        sum = sum + (pow(digit, numberOfDigits));
        temp /= 10;
    }

    if (sum == number) {
        cout << number << " is AN ARMSTRONG NUMBER!!" << endl;
    } else {
        cout << number << " is NOT AN ARMSTRONG NUMBER!!!!" << endl;
    }

    return 0;
}